app/realm/writers/client_upgrade.js

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {combineWriters} from 'realm-react-redux';
+import {ClientUpgradeTypes} from 'app/realm/action_types';
+
+function clientUpgrade(realm, action) {
```

Comment:
Why do we need this?

Reply:
The writer is what creates/update the Realm DB table, in this case 'ClientUpgrade'. There is an issue here as I mentioned above where I was not assigning the primary ID on the model so in this case I need to directly call out the ID. I have the overwrite set as true as we always just want to update the one item.

Reply:
same question